curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e042661ad2663f0d468740.md

```diff
@@ -15,7 +15,10 @@ Use object destructuring to access the `coachName` value.
 
 ```js
 
-assert.match(code, /const\s*{\s*coachName\s*}\s*=\s*myFavoriteFootballTeam\.headCoach\s*/)
+assert.match(code,
+/const\s*{\s*coachName\s*}\s*=\s*myFavoriteFootballTeam\.headCoach\s*/ || 
+/const\s*{\s*headCoach\s*:\s*{\s*coachName\s*}\s*}\s*=\s*myFavoriteFootballTeam\s*/
+)
```

Comment:
This won't work like this. The `||` is a logical OR operator, it will return the _thing_ on the left if it's not falsy, otherwise it returns the second _thing_. This means for this case always will be checked `/const\s*{\s*coachName\s*}\s*=\s*myFavoriteFootballTeam\.headCoach\s*/` regex.

```suggestion
assert.match(code, /const\s*{\s*(coachName\s*}\s*=\s*myFavoriteFootballTeam\.headCoach|headCoach\s*:\s*{\s*coachName\s*}\s*}\s*=\s*myFavoriteFootballTeam)\s*;?/)
```